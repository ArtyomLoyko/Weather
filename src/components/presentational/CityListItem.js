import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Item = styled.li`
  display: flex;
  justify-content: flex-start;
  padding: 5px;
  flex-wrap: wrap;

  &:not(:last-child) {
    border-bottom: 1px solid rgba(0, 0, 0, 0.15);
    cursor: pointer;
  }

  &:first-child {
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
  }

  &:last-child {
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
  }

  &:hover {
    background: rgba(0, 0, 0, 0.15);
  }
`;

export const CityListItem = ({ data, onClick }) => (
  <Item onClick={onClick}>{data}</Item>
);

CityListItem.propTypes = {
  data: PropTypes.string.isRequired,
};

export default CityListItem;
