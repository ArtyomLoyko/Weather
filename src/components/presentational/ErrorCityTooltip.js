import React from 'react';
import styled from 'styled-components';
import onClickOutside from 'react-onclickoutside';

const Wrapper = styled.div`
  position: fixed;
  top: 15px;
  left: 45%;
  font-size: 1.3rem;
  z-index: 3;
  border-radius: 5px;
  background-color: rgba(255, 0, 0, 0.65);
  padding: 10px;
  border: 1px solid rgba(0, 0, 0, 0.15);
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
`;

class ErrorCityTooltip extends React.Component {
  handleClickOutside = () => this.props.toggleWarning();

  render() {
    return <Wrapper>Incorrect city!</Wrapper>;
  }
}

export default onClickOutside(ErrorCityTooltip);
