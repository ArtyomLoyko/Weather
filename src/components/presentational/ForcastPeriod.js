import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin: 0 15px;
`;

const Icon = styled.img`
    width: 75%;
    height: 75%;
    display: flex;
    margin: auto;
`;

const Description = styled.span`
    font-size: 0.7rem;
    padding: 0 0 3px;
`;

const Temp = styled.strong`
    padding: 2px;
`;

const Value = styled.span`
    font-size: 0.8rem;
    padding: 1.5px;
`;

export const ForcastPeriod = ({ weatherPeriod }) => {
    const { dt_txt, weather, main, wind } = weatherPeriod[0];
    return (
        <Wrapper>
            <strong>{dt_txt}</strong>
            <Icon
                src={`http://openweathermap.org/img/w/${weather[0].icon}.png`}
                alt={weather[0].description}
            />
            <Description>{weather[0].description}</Description>
            <Temp>{main.temp} Â°C</Temp>
            <Value>{wind.speed} m/s</Value>
            <Value>{main.pressure} hPa</Value>
            <Value>{main.humidity} %</Value>
        </Wrapper>
    );
};

ForcastPeriod.propTypes = {
    weatherPeriod: PropTypes.array.isRequired,
};

export default ForcastPeriod;
