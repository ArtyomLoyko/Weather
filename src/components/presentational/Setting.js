import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import styled from 'styled-components';

import './../../styles/FormInput.css';

const Wrapper = styled.form`
    position: fixed;
    top: 15px;
    left: 15px;
    background-color: white;
    z-index: 2;
    font-family: sans-serif;
    text-align: center;
    border-radius: 5px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    box-shadow: 0 3px 5px rgba(0, 0, 0, 0.2);

    :hover {
        box-shadow: 0 4px 6px rgba(0, 0, 0, 0.4);
    }
`;

const Text = styled.h1`
    font-size: 1.5rem;
`;

const FieldContainer = styled.div`
    display: flex;
    margin: 0 12px 15px;
`;

const RadioLabel = styled.label`
    display: flex;
    margin-bottom: 5px;
    cursor: pointer;
`;

const SelectContainer = styled.div`
    display: flex;
    flex-direction: row;
`;

const RadioLabelsContainer = styled.div`
    display: flex;
    margin: 1.5px 0 0 5px;
    flex-direction: column;
`;

const SubmitButton = styled.button`
    width: 150px;
    height: 40px;
    display: flex;
    justify-content: center;
    font-size: 18px;
    background-color: white;
    margin: 10px auto 15px;
    outline: none;
    border-radius: 5px;
    cursor: pointer;
    border: 1px solid rgba(0, 0, 0, 0.15);
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
    user-select: none;

    :hover {
        background-color: rgba(240, 240, 240, 0.8);
    }
`;

export const Setting = ({ handleSubmit }) => {
    return (
        <Wrapper onSubmit={handleSubmit}>
            <Text>WEATHER FORECAST</Text>
            <FieldContainer>
                <label>
                    Insert City:
                    <Field
                        name="city"
                        component="input"
                        className="form-input"
                        type="text"
                        placeholder="city *"
                        required
                    />
                </label>
            </FieldContainer>
            <FieldContainer>
                <SelectContainer>
                    <span>Select Day:</span>
                    <RadioLabelsContainer>
                        <RadioLabel>
                            <Field
                                name="day"
                                component="input"
                                className="form-radio"
                                type="radio"
                                value="today"
                                required
                            />
                            Today
                        </RadioLabel>
                        <RadioLabel>
                            <Field
                                name="day"
                                component="input"
                                className="form-radio"
                                type="radio"
                                value="tomorrow"
                                required
                            />
                            Tomorrow
                        </RadioLabel>
                        <RadioLabel>
                            <Field
                                name="day"
                                component="input"
                                className="form-radio"
                                type="radio"
                                value="day after tomorrow"
                                required
                            />
                            Day after tomorrow
                        </RadioLabel>
                    </RadioLabelsContainer>
                </SelectContainer>
            </FieldContainer>
            <SubmitButton type="submit">Show Weather</SubmitButton>
        </Wrapper>
    );
};

Setting.propTypes = {
    handleSubmit: PropTypes.func.isRequired,
};

export default reduxForm({
    form: 'setting',
    destroyOnUnmount: false,
})(Setting);
