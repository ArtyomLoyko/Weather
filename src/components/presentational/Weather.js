import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import ForcastPeriod from './ForcastPeriod';

const Wrapper = styled.section`
    display: flex;
    flex-direction: column;
    justify-content: center;
    text-align: center;
    font-family: sans-serif;
`;

const Forecast = styled.div`
    display: flex;
    background-color: rgba(255, 238, 0, 0.4);
    padding: 15px;
`;

export const Weather = ({ city, weather }) => {
    return (
        <Wrapper>
            <h2>{city}</h2>
            <Forecast>
                {weather.map(weatherPeriod => (
                    <ForcastPeriod
                        key={weatherPeriod[0].dt}
                        weatherPeriod={weatherPeriod}
                    />
                ))}
            </Forecast>
        </Wrapper>
    );
};

Weather.propTypes = {
    city: PropTypes.string.isRequired,
    weather: PropTypes.array.isRequired,
};

export default Weather;
