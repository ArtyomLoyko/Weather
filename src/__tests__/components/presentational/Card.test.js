import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './../../../components/presentational/Card';

describe('Card', () => {
    const mockHandleClick = jest.fn();

    const props = {
        isTurned: false,
        isOut: false,
        scirt: 'some/path',
        image: 'some/path',
        handleClick: mockHandleClick,
    };

    const card = shallow(<Card {...props} />);

    it('renders properly', () => {
        expect(card).toMatchSnapshot();
    });
});
