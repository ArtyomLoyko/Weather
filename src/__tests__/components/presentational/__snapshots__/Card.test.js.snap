// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    handleClick={[Function]}
    image="some/path"
    isOut={false}
    isTurned={false}
    scirt="some/path"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="card"
        src="some/path"
/>,
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "card",
        "src": "some/path",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="card"
          src="some/path"
/>,
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "card",
          "src": "some/path",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
